src/zope/interface/interface.py

```diff
@@ -154,18 +154,7 @@ def __adapt__(self, obj):
                 return adapter
 
 
-InterfaceBase = InterfaceBasePy
-try:
-    from zope.interface._zope_interface_coptimizations import InterfaceBase
-except ImportError:
-    pass
-
-
-adapter_hooks = []
-try:
-    from zope.interface._zope_interface_coptimizations import adapter_hooks
-except ImportError:
-    pass
+adapter_hooks = _use_c_impl([], 'adapter_hooks')
```

Comment:
Maybe it would be good for these modules to define `__all__`?

Another good followup PR would be to standardize the convention used for defining the Python fallback so we only have to write one extra attribute (and which is consistent).

Also, a good followup PR might be to make Sphinx's `conf.py` define `PURE_PYTHON` so that the objects that are documented could be written with their preferred name, instead of like `.. autofunction:: implementedByFallback`. (Ok, I only actually see that once.)